scripts/validate_config.py

```diff
@@ -59,25 +56,17 @@ def confirm_environment_valid(self):
 
     def confirm_onion_config_valid(self):
         """
-        We support both v2 and v3 onion services, so if the values
-        in the config file match either format, the configuration is considered
-        valid. A deprecation warning is shown if v2 services are in use.
+        Only v3 onion services are supported.
         """
-        v2_config = False
         try:
             self.confirm_onion_v3_url()
             self.confirm_onion_v3_auth()
         except AssertionError:
-            self.confirm_onion_v2_url()
-            self.confirm_onion_v2_auth()
-            v2_config = True
-
-        if v2_config:
             print(
-                "WARNING: v2 onion service configuration found.\n"
-                "Support for v2 onion services will be removed from SecureDrop in February 2021.\n"
-                "Migration guide: https://securedrop.org/v2-onion-eol/"
+                "ERROR: Onion service configuration missing or does not match expected format.\n"
+                "Please note that only v3 onion services are supported.\n"
             )
+            raise
```

Comment:
IMO this script shouldn't display tracebacks during expected validation errors, but that's the current pattern. See #683 as well for suggested refactoring of this whole logic.